### PR TITLE
wix-ui-core: <Popover> - Fix attachTo window

### LINKED ISSUE #523: wix-ui-core: <Popover> - Multiple Popover CSS BUG
Found a bug in Popover.
If you have multiple Popovers in a page,
In this case both attach to the window (I didn't test other attach strategies),
and you close one Popover, the the other Popover's css is gone !

I think it has something to do with our hack to apply/remove css from a portal.

To see the bug in action:
Open Popover's story (in this PR's storybook),
and under the examples part, click the button saying "Click to close 1"

See #519

### PR DESCRIPTION
**Pull request by hepiyellow**
Fixes #523 

Also fixes issue with React 15, where setting the attachTo to `document.body` is casing the popover to overwrite the `body` instead of appending to it.

### CODE REVIEW

File: packages/wix-ui-core/src/components/Popover/Popover.tsx [hunk 1]
```diff
@@ -260,18 +248,46 @@ export class Popover extends React.Component<PopoverType, PopoverState> {
...
+    this.appendToNode = this.getOrCreateAppendToNode({appendTo, targetRef: this.targetRef});
+    // TODO: I think we should throw an error is this.appendToNode is undefined, since 
+    // react-portal will attach it to a new default div under document.body,
+    // and we could not add the stylable styles to that div.
     if (this.appendToNode) {
-      this.stylesObj = style('root', {}, props);
-      this.applyStylesToAppendedNode(props);
+      // Why do we do this here ?(in componentDidMount and not ONLY in render? or when we actually attachStylesToNode)
+      this.stylesObj = style('root', {}, this.props);
+      // TODO: remove this, it is called in render
```
Comment (hepiyellow) [inline packages/wix-ui-core/src/components/Popover/Popover.tsx]: @shlomitc do you agree?

**Comment by alisey** [comment]
Can you please hold off merging, I have #539 which affects all the tests to make them compatible with the browser environment, want to merge them asap before people start changing more tests.

**Comment by hepiyellow** [comment]
@alisey This is quite urgent, are you merging today?

File: packages/wix-ui-core/src/components/Popover/Popover.spec.tsx [hunk 2]
```diff
@@ -80,97 +81,129 @@ describe('Popover', () => {
 
   describe('appendTo', () => {
     let wrapper, driver;
+    const createDriverFromWrapper = (wrapper2: ReactWrapper) => popoverDriverFactory({element: wrapper2.children().at(0).getDOMNode(), eventTrigger: null});
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/Popover/Popover.spec.tsx]: wrapper2 is an aweful name

File: packages/wix-ui-core/src/components/Popover/Popover.tsx [hunk 3]
```diff
@@ -163,33 +164,19 @@ export class Popover extends React.Component<PopoverType, PopoverState> {
 
   static propTypes = {
     className: string,
-    /** The location to display the content */
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/Popover/Popover.tsx]: let's not remove the comments, it is still in use by the `wix-ui-adi` Notification component (not in build time)

File: packages/wix-ui-core/src/components/Popover/Popover.tsx [hunk 4]
```diff
@@ -223,14 +210,14 @@ export class Popover extends React.Component<PopoverType, PopoverState> {
     return this.wrapWithAnimations(popper);
   }
 
-  applyStylesToAppendedNode(props) {
-    const {shown} = props;
-    const shouldAnimate = shouldAnimatePopover(props);
+  applyStylesToAppendedNode() {
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/Popover/Popover.tsx]: `applyStylesToPortaledNode`

File: packages/wix-ui-core/src/components/Popover/Popover.tsx [hunk 5]
```diff
@@ -255,27 +242,38 @@ export class Popover extends React.Component<PopoverType, PopoverState> {
...
+    const {appendTo} = this.props;
+    this.appendToNode = getAppendToNode({appendTo, targetRef: this.targetRef});
     if (this.appendToNode) {
-      this.stylesObj = style('root', {}, props);
-      this.applyStylesToAppendedNode(props);
+      this.portalNode = document.createElement('div');
+      this.appendToNode.appendChild(this.portalNode);
+      // Why do we do this here ?(in componentDidMount and not ONLY in render? or when we actually attachStylesToNode)
+      this.stylesObj = style('root', {}, this.props);
+      // TODO: remove this, it is called in render
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/Popover/Popover.tsx]: you can try removing it but I'm not sure what will be the effect of it

File: packages/wix-ui-core/src/components/Popover/Popover.spec.tsx [hunk 6]
```diff
@@ -80,97 +81,129 @@ describe('Popover', () => {
 
   describe('appendTo', () => {
     let wrapper, driver;
+    const createDriverFromWrapper = (_wrapper: ReactWrapper) => popoverDriverFactory({element: _wrapper.children().at(0).getDOMNode(), eventTrigger: null});
```
Comment (alisey) [inline packages/wix-ui-core/src/components/Popover/Popover.spec.tsx]: In the master branch we have the following, which basically does what you need:
https://github.com/wix/wix-ui/blob/master/packages/wix-ui-core/src/components/Popover/Popover.spec.tsx#L19

**Comment by alisey** [comment]
Sorry, but the tests are now completely unreadable. The control flow is all over the place, and from looking at a single test I cannot tell what it's doing, and whether it's doing it correctly. Verbosity is not a bad thing for tests.

And why did you have to change classList back to className? `expect(getExpectedContentParent().className).not.toBe(styles.root);` - this doesn't test anything, the class name could be `root groot`, and the test would not catch that.

**Comment by hepiyellow** [comment]
OK:
- I removed the first `afterEach` which did the `unmount` and did is verbosely in each test. Maybe that simplifies the fact that there are 2 `afterEach`'s.
- I re-added the use of `classList` - I missed that in the conflict resolution. Thanks !

See PR: https://github.com/wix/wix-ui/pull/545

**Pull request merged**